waspc/data/Generator/templates/react-app/vite/detectServerImports.ts

```diff
@@ -0,0 +1,63 @@
+{{={= =}=}}
+import { type Plugin } from 'vite'
+import path from 'path'
+
+const waspProjectDirAbsPath = getWaspProjectDirAbsPathFromCwd()
+
+export function detectServerImports(): Plugin {
+  return {
+    name: 'wasp-detect-server-imports',
+    enforce: 'pre',
+    resolveId(source, importer) {
+      if (!importer) {
+        return
+      }
+
+      const importerRelativePath = getPathRelativeToWaspProjectDir(importer)
+
+      if (!isPathToUserCode(importerRelativePath)) {
+        return
+      }
```

Comment:
To go a step further with the "Parse, don't validate" philosohpy, we could do this:
```typescript
type RelativePathToUserCode = string & { _brand: 'relativePathToUserCode' }

// I used null, but you can also create a Result type
function parsePathToUserCode(importerRelativePath: RelativePath): RelativePathToUserCode | null

// ...

function ensureNoServerImports(source: string, relativeImporter: RelativePathToUserCode): void
``` 
This "ensures" (module type assertions) we can never call `ensureNoServerImports` without first verifying that we're dealing with user code.